### ISSUE TITLE
If you build with trimming enabled, future builds with it disabled will produce incorrect output

### ISSUE THREAD

**Issue by kg** [dotnet/runtime#53612]
### Description

cc @radical 

If I do an incremental build with ```EnableAggressiveTrimming=true```, then do another one later with it ```=false``` after changing some .cs files, not everything is rebuilt so you get outputs that have type load failures and such, like so:

```fail: console.error: Unhandled exception in _finalize_startup Error: System.TypeLoadException: Could not load type of field 'System.Runtime.InteropServices.JavaScript.Runtime:_boundObjects' (0) due to: Could not resolve type with token 01000012 from typeref```

Obliterating my artifacts folder and then doing a full build is usually the fix for this.

Will come back and paste in my exact build command lines later once my rebuild is done.

### Configuration

dotnet/runtime/main current as of today

### Regression?

No

**Comment by lewing** [comment]
@radical status on this?

**Comment by radical** [comment]
This issue is with local `dotnet/runtime` builds. Lower priority as compared to other issues, so haven't had a chance to look at this.

I'm working on incremental builds now, so after that we can see if this becomes a problem for user projects.

**Comment by lewing** [comment]
@radical is this fixed?

**Comment by radical** [comment]
> @radical is this fixed?

No, this will be fixed after the publish->build changes. This one will need to copy the untrimmed assemblies to a single path, and use that for mono-aot-cross.

**Comment by radical** [comment]
Should be fixed by #58913 .
